core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Boxable.java

```diff
@@ -18,20 +23,26 @@
 public interface Boxable extends Nameable, Barcodable, SecurableByProfile, Serializable {
 
   public enum EntityType {
-    SAMPLE(SampleImpl.class),
-    LIBRARY(LibraryImpl.class),
-    DILUTION(LibraryDilution.class),
-    POOL(PoolImpl.class);
+    SAMPLE(SampleImpl.class, SampleBoxPosition::new),
+    LIBRARY(LibraryImpl.class, LibraryBoxPosition::new),
+    DILUTION(LibraryDilution.class, DilutionBoxPosition::new),
+    POOL(PoolImpl.class, PoolBoxPosition::new);
 
     private final Class<? extends Boxable> persistClass;
+    private final Supplier<? extends AbstractBoxPosition> positionConstructor;
 
-    private EntityType(Class<? extends Boxable> persistClass) {
+    private EntityType(Class<? extends Boxable> persistClass, Supplier<? extends AbstractBoxPosition> positionConstructor) {
       this.persistClass = persistClass;
+      this.positionConstructor = positionConstructor;
     }
 
     public Class<? extends Boxable> getPersistClass() {
       return persistClass;
     }
+
+    public AbstractBoxPosition makeBoxPosition() {
+      return positionConstructor.get();
+    };
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)